shiny/experimental/ui/_sidebar.py

```diff
@@ -163,44 +157,100 @@ def layout_sidebar(
+        ),
+        xmlns="http://www.w3.org/2000/svg",
+        viewBox="0 0 16 16",
+        class_="bi bi-chevron-down collapse-icon",
+        style="fill:currentColor;",
+        aria_hidden="true",
+        role="img",
+    )
+
+
+def sidebar_init_js() -> Tag:
+    # Note: if we want to avoid inline `<script>` tags in the future for
+    # initialization code, we might be able to do so by turning the sidebar layout
+    # container into a web component
     return tags.script(
-        {"data_bslib_sidebar_init": True},
-        """
-        var thisScript = document.querySelector('script[data-bslib-sidebar-init]');
-        thisScript.removeAttribute('data-bslib-sidebar-init');
-
-        // If this layout is the innermost layout, then allow it to add CSS
-        // variables to it and its ancestors (counting how parent layouts there are)
-        var thisLayout = $(thisScript).parent();
-        var noChildLayouts = thisLayout.find('.bslib-sidebar-layout').length === 0;
-        if (noChildLayouts) {
-        var parentLayouts = thisLayout.parents('.bslib-sidebar-layout');
-        // .add() sorts the layouts in DOM order (i.e., innermost is last)
-        var layouts = thisLayout.add(parentLayouts);
-        var ctrs = {left: 0, right: 0};
-        layouts.each(function(i, x) {
-            $(x).css('--bslib-sidebar-counter', i);
-            var right = $(x).hasClass('sidebar-right');
-            $(x).css('--bslib-sidebar-overlap-counter', right ? ctrs.right : ctrs.left);
-            right ? ctrs.right++ : ctrs.left++;
-        });
-        }
-
-        // If sidebar is marked open='desktop', collapse sidebar if on mobile
-        if (thisLayout.data('sidebarInitAutoCollapse')) {
-        var initCollapsed = thisLayout.css('--bslib-sidebar-js-init-collapsed');
-        if (initCollapsed === 'true') {
-            thisLayout.addClass('sidebar-collapsed');
-            thisLayout.find('.collapse-toggle').attr('aria-expanded', 'false');
-        }
-        }
-        """,
+        {"data-bslib-sidebar-init": True},
+        HTML("bslib.Sidebar.initCollapsibleAll()"),
     )
+
+
+########################################################
+
+
+def panel_sidebar(
+    *args: TagChild | TagAttrs,
+    width: int = 4,
+    **kwargs: TagAttrValue,
+) -> Sidebar:
+    """Deprecated. Please use `ui.sidebar()` instead of `ui.panel_sidebar()`."""
+    warn_deprecated("`panel_sidebar()` is deprecated. Please use `sidebar()` instead.")
+    return sidebar(
+        *args,
+        # id="bslib-panel-sidebar",
+        width=f"{int(width / 12 * 100)}%",
+        **kwargs,
+    )
+
+
+def panel_main(
+    *args: TagChild | TagAttrs,
+    width: int = 8,
+    **kwargs: TagAttrValue,
+) -> TagList:
+    """Deprecated. Please supply `*args` directly to `layout_sidebar()` instead."""
+    warn_deprecated(
+        "`panel_main()` is deprecated. Please supply `*args` directly to `layout_sidebar()` instead."
+    )
```

Comment:
Because this change requires the user to not just rename a function, but also restructure their code, we need to think about a good strategy for doing this.

How about:
- No deprecation messages today, and existing code just works.
- Change all examples to use the new API.
- In, say, 6 months, start emitting messages for code that uses the old API.